modules/actions/pool/sync-join-exits.ts

```diff
@@ -0,0 +1,116 @@
+import { Chain, PoolEventType } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+import { V3VaultSubgraphClient } from '../../sources/subgraphs';
+import { formatUnits } from 'viem';
+import { JoinExit_OrderBy, OrderDirection } from '../../sources/subgraphs/balancer-v3-vault/generated/types';
+import { roundToHour } from '../../common/time';
+
+/**
+ * Get the join and exit events from the subgraph and store them in the database
+ *
+ * @param vaultSubgraphClient
+ */
+export const syncJoinExits = async (vaultSubgraphClient: V3VaultSubgraphClient, chain: Chain): Promise<string[]> => {
+    const vaultVersion = 3;
+
+    // Get latest event from the DB
+    const latestEvent = await prisma.poolEvent.findFirst({
+        where: {
+            type: {
+                in: ['JOIN', 'EXIT'],
+            },
+            chain: chain,
+            vaultVersion,
+        },
+        orderBy: {
+            blockNumber: 'desc',
+        },
+    });
+
+    // Get events
+    const { joinExits } = await vaultSubgraphClient.JoinExits({
+        first: 1000,
+        where: {
+            blockNumber_gte: String(latestEvent?.blockNumber || 0),
+        },
+        orderBy: JoinExit_OrderBy.BlockNumber,
+        orderDirection: OrderDirection.Asc,
+    });
+
+    // Store only the events that are not already in the DB
+    const existingEvents = await prisma.poolEvent.findMany({
+        where: {
+            id: { in: joinExits.map((event) => event.id) },
+            type: {
+                in: ['JOIN', 'EXIT'],
+            },
+            chain: chain,
+            vaultVersion,
+        },
+    });
+
+    const events = joinExits.filter((event) => !existingEvents.some((existing) => existing.id === event.id));
+
+    // Prepare DB entries
+    const dbEntries = await Promise.all(
+        events.map(async (event) => {
```

Comment:
can you regfactor this to how swaps are synced? No reason there are written so differently. Swaps sync seems super clean compared to this...